core/src/test/java/io/github/tcdl/msb/message/MessageFactoryTest.java

```diff
@@ -219,4 +220,28 @@ public void testCreateAckBuilder() throws Exception {
         Acknowledge ack = messageFactory.createAckBuilder().build();
         assertNotNull(ack.getResponderId());
     }
+
+    @Test
+    public void testCreateRequestMessageBuilderPublishedAtPresent() {
+        String bodyText = "body text";
+        Payload requestPayload = TestUtils.createPayloadWithTextBody(bodyText);
+
+        Builder requestMessageBuilder = TestUtils.createMessageBuilder();
+
+        Message message = messageFactory.createRequestMessage(requestMessageBuilder, requestPayload);
+
+        assertNotNull(message.getMeta().getPublishedAt());
```

Comment:
This test seems to be weaker than it could. Could we use fixed clock and check for exact value rather than verify nullability?